### PR TITLE
Finish some todo tasks

### PR DESCRIPTION
**Pull request by karangattu**
Address `TODO-karan` tasks.
Also, has a POC of how to check if new UI methods are added without making sure they have module support

### CODE REVIEW

File: tests/e2e/bugs/0696-resolve-id/check.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    input_switch=False,
+    input_text="text0",
+    input_text_area="text_area0",
+    navset_bar="a",
+    navset_card_pill="a",
+    navset_card_tab="a",
+    navset_hidden="a",
+    navset_pill="a",
+    navset_tab="a",
+)
```
Comment (schloerke) [inline tests/e2e/bugs/0696-resolve-id/check.py:76]: I like this idea and where it's going. 

- [ ] Move `test_0696_resolve_id.py`'s custom expectation method into a new file (`mod_state.py`?) that can be imported into here and the original file. 
- [ ] Change class name to `ModState`
- [ ] Have expectation methods receive a `ModState` object. 
- [ ] Double check if a value is missing when creating a NamedTuple, an error is thrown.
- [ ] Have a `default_mod_state` object (which currently equals `component` above).
- [ ] Is it possible to import `mod_state` into the app to retrieve the `input_keys` and `x_input_keys`?

File: tests/e2e/controls.py [hunk 2]
```diff
@@ -2670,6 +2688,27 @@ def expect_panels(
...
 
+    def set(
+        self,
+        selected: str | list[str],
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        if isinstance(selected, str):
+            selected = [selected]
+        # self.loc resolves to 4 elements, thus violating the strict mode requirement
```
Comment (schloerke) [inline tests/e2e/controls.py:2699]:
```suggestion
        # self.loc resolves to multiple elements, thus violating the strict mode requirement
```

File: tests/e2e/controls.py [hunk 3]
```diff
@@ -2670,6 +2688,27 @@ def expect_panels(
...
+        # self.loc.wait_for(state="visible", timeout=timeout)
+        # self.loc.scroll_into_view_if_needed(timeout=timeout)
```
Comment (schloerke) [inline tests/e2e/controls.py:2701]: Can these be done on `element` in the next line?

File: tests/e2e/controls.py [hunk 4]
```diff
@@ -2717,3 +2758,404 @@ def expect_icon(self, value: PatternOrStr, *, timeout: Timeout = None) -> None:
...
+        if self.loc_body_visible.count() != int(open):
+            self.toggle(timeout=timeout)
+
+    def toggle(self, *, timeout: Timeout = None) -> None:
+        self.loc.wait_for(state="visible", timeout=timeout)
+        self.loc.scroll_into_view_if_needed(timeout=timeout)
+        self.loc_header.click(timeout=timeout)
+
+
+class _OverlayBase(_InputBase):
```
Comment (schloerke) [inline tests/e2e/controls.py:2774]: Merge in the latest from `main`. Hopefully most of the new code in the rest of the file is not double added. 

... unless there are new changes?

File: tests/e2e/controls.py [hunk 5]
```diff
@@ -816,9 +816,13 @@ def __init__(
         )
 
     def set(self, value: bool, *, timeout: Timeout = None, **kwargs: object) -> None:
+        self.loc.wait_for(state="visible", timeout=timeout)
```
Comment (schloerke) [inline tests/e2e/controls.py:819]: Please remove comment added in #728 about adding these lines throughout controls

**Review by schloerke — CHANGES_REQUESTED** [review]

File: tests/e2e/inputs/input_file/app.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+app_ui = ui.page_fluid(
+    ui.input_file("file1", "Choose CSV File", accept=[".csv"], multiple=False),
+    ui.input_checkbox_group(
+        "stats",
+        "Summary Stats",
+        choices=["Row Count", "Column Count", "Column Names"],
+        selected=["Row Count", "Column Count", "Column Names"],
+    ),
+    ui.output_table("summary"),
+    ui.input_file("file2", "Choose CSV File", accept=[".csv"], multiple=True),
```
[the remark this answers is not in the document]
Comment (karangattu) [reply tests/e2e/inputs/input_file/app.py:15]: As far I recollect, one of the challenges was when uploading multiple files, it did not show an array instead it just shows `n` files uploaded where `n` is the number of files that the user selected.

[the remark this answers is not in the document]
Comment (karangattu) [reply tests/e2e/inputs/input_file/app.py:15]: I do check that `expect(file2.loc_file_display).to_have_value("2 files")`

File: tests/e2e/bugs/0696-resolve-id/check.py (hunk 1 above)
Comment (schloerke) [reply tests/e2e/bugs/0696-resolve-id/check.py]: Turning into comment. 

Resolving

**Pull request merged**